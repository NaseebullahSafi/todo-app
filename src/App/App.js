import React from "react";

import { GlobalStyles } from "../global.style";
import Header from "src/organisms/Header";
import Main from "src/organisms/Main";
import { StyledApp } from "./App.style";

/**
 * App component - is the top most root component
 * which wraps everything. inc. global styles.
 */
const App = () => (
  <StyledApp>
    <GlobalStyles />
    <Header />
    <Main />
  </StyledApp>
);

export default App;
