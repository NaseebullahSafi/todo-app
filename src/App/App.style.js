import styled from "styled-components";

export const StyledApp = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
`;
