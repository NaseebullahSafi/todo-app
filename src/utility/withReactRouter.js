import React from "react";
import { MemoryRouter } from "react-router-dom";
import { create } from "react-test-renderer";

export const withReactRouter = (component) =>
  create(<MemoryRouter>{component}</MemoryRouter>);
