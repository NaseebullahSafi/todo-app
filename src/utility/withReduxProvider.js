import React from "react";
import { Provider } from "react-redux";

import store from "../Store";

export const ReduxProvider = ({ children }) => (
  <Provider store={store}>{children}</Provider>
);
