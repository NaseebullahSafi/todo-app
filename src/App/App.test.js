import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import { withReactRouter } from "src/utility/withReactRouter";
import App from "./App";

describe("App", () => {
  it("renders correctly", () => {
    const tree = withReactRouter(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
