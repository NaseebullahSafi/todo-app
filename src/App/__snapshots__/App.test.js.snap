// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
.c1 {
  width: 100%;
}

.c3 a {
  background-color: #fff;
  border-radius: 0.25rem;
  box-shadow: 0 4px 6px rgba(50,50,93,0.11),0 1px 3px rgba(0,0,0,0.08);
  color: #525f7f;
  font-size: 0.875rem;
  font-weight: 500;
  min-width: 145px;
  padding: 1rem;
  -webkit-transition: all 0.15s ease;
  transition: all 0.15s ease;
}

.c3 a.selected {
  background-color: #5e72e4;
  color: white;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  list-style: none;
  margin: 3em 0 5em;
  padding-left: 0;
}

.c2 a {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 a svg {
  margin-right: 9px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 .fade-enter {
  opacity: 0.01;
}

.c4 .fade-enter.fade-enter-active {
  opacity: 1;
  -webkit-transition: opacity 300ms ease-in;
  transition: opacity 300ms ease-in;
}

.c4 .fade-exit {
  opacity: 1;
}

.c4 .fade-exit.fade-exit-active {
  opacity: 0.01;
  -webkit-transition: opacity 300ms ease-in;
  transition: opacity 300ms ease-in;
}

.c4 div.transition-group {
  position: relative;
}

.c5 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  max-width: 800px;
}

.c6 {
  position: fixed;
  bottom: 0;
  right: 0;
  margin: 1em;
  border-radius: 100%;
  background-color: #f5365c;
  padding: 1em;
  width: 70px;
  height: 70px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-shadow: 0 4px 6px rgba(50,50,93,0.11),0 1px 3px rgba(0,0,0,0.08);
  -webkit-transition: all 0.15s ease-in;
  transition: all 0.15s ease-in;
  cursor: pointer;
  color: white;
}

.c6 svg {
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
}

.c6:hover {
  background-color: #5e72e4;
}

.c6:hover svg {
  color: white;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media screen and (min-width:300px) {
  .c1 {
    max-width: 300px;
  }
}

@media screen and (min-width:400px) {
  .c1 {
    max-width: 400px;
  }
}

@media screen and (min-width:600px) {
  .c1 {
    max-width: 600px;
  }
}

@media screen and (min-width:200px) {
  .c3 {
    margin: 0.5em;
  }
}

@media screen and (min-width:200px) {
  .c2 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media screen and (min-width:400px) {
  .c2 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media screen and (min-width:300px) {
  .c5 {
    width: 100vw;
  }
}

<div
  className="c0"
>
  <header
    className="c1"
  >
    <ul
      className="c2"
    >
      <li
        className="c3"
      >
        <a
          aria-current={null}
          href="/todos"
          onClick={[Function]}
        >
          <svg
            className="feather feather-layers"
            fill="none"
            height="16"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="12 2 2 7 12 12 22 7 12 2"
            />
            <polyline
              points="2 17 12 22 22 17"
            />
            <polyline
              points="2 12 12 17 22 12"
            />
          </svg>
          Todos
        </a>
      </li>
      <li
        className="c3"
      >
        <a
          aria-current={null}
          href="/create"
          onClick={[Function]}
        >
          <svg
            className="feather feather-plus-square"
            fill="none"
            height="16"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="18"
              rx="2"
              ry="2"
              width="18"
              x="3"
              y="3"
            />
            <line
              x1="12"
              x2="12"
              y1="8"
              y2="16"
            />
            <line
              x1="8"
              x2="16"
              y1="12"
              y2="12"
            />
          </svg>
          Create
        </a>
      </li>
      <li
        className="c3"
      >
        <a
          aria-current={null}
          href="/play"
          onClick={[Function]}
        >
          <svg
            className="feather feather-play"
            fill="none"
            height="16"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="5 3 19 12 5 21 5 3"
            />
          </svg>
          Play
        </a>
      </li>
    </ul>
  </header>
  <main
    className="c4"
  >
    <div>
      <section
        className="c5"
      />
    </div>
    <button
      className="c6"
      onClick={[Function]}
    >
      <svg
        className="feather feather-play-circle"
        fill="none"
        height="30"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width="30"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <polygon
          points="10 8 16 12 10 16 10 8"
        />
      </svg>
    </button>
  </main>
</div>
`;
